core/src/main/java/org/elasticsearch/search/SearchExtParser.java

```diff
@@ -20,12 +20,19 @@
 package org.elasticsearch.search;
 
 import org.elasticsearch.common.xcontent.XContentParser;
-import org.elasticsearch.search.internal.SearchContext;
 
 /**
- *
+ * Parser for the ext section of a search request, which can hold custom fetch sub phases config
  */
-public interface SearchParseElement {
+public interface SearchExtParser {
+
+    /**
+     * Returns the name of the element that this parser is able to parse
+     */
+    String getName();
 
-    void parse(XContentParser parser, SearchContext context) throws Exception;
+    /**
+     * Parses the element whose name is returned by {@link #getName()}
+     */
+    Object parse(XContentParser parser) throws Exception;
```

Comment:
potentially, but I will remove the parse field matcher very soon, so I would not add it here.